### ISSUE TITLE
Major problems with CLJS repl setup

### ISSUE THREAD

**Issue by baflo** [BetterThanTomorrow/calva#216]
Hi, I'm a very beginner with Clojure (so excuse me if I don't get it right due to configuration) and trying to set up Calva in VSCode . After a lot of other attempts, I followed the strategy [creating a reagent project](https://github.com/BetterThanTomorrow/calva/wiki/Connecting-To-a-Running-REPL-Server#from-scratch-using-the-reagent-leiningen-template).

Without touching any of the configuration, I'm not able to attach a valid ClojureScript REPL session. 

- When I try to "Open a ClojureScript REPL" I get errors like `cannot read property 'client' from null` or similar. 
- When I try to `Start a project REPL and connect` with "Leiningen" or "Leiningen Figwheel" I get the folling errors in the task shell:

```
> Executing task: C:\Users\fbachmann\bin\lein.bat update-in :dependencies conj "[nrepl \"0.6.0\"]" -- update-in :plugins conj "[cider/cider-nrepl \"0.21.1\"]" -- update-in "[:repl-options :nrepl-middleware]" conj "[\"cider.nrepl/cider-middleware\"]" -- with-profile dev repl :headless <

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by mranderson048.orchard.v0v3v0.dynapath.v0v2v5.dynapath.defaults$eval1958$fn__1959 to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of mranderson048.orchard.v0v3v0.dynapath.v0v2v5.dynapath.defaults$eval1958$fn__1959
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Warning: implicit hook found: lein-environ.plugin/hooks
Hooks are deprecated and will be removed in a future version.
java.lang.RuntimeException: EOF while reading
```
- I also tried to attach to an existing REPL: Using Leiningen, I start `lein repl` in the project, then do `(user/start-fw ) (user/cljs )`, which results in a usable CLJS REPL. I am able to connect to this repl in VSCode `1.33` , but not in `1.36 Insiders` (which just says "Failed connecting. (Calva needs a REPL started before it can connect.)").

However, even in the connected REPL, I cannot evaluate the cljs:
- Using `evaluate current form (or selection)` I get the error `cannot read property 'eval' of null`.
- `Attach to a ClojureScript session` practically does nothing except printing `Connecting to Figwheel`

My host is a Windows System, I also tried in an Ubuntu VM and in Ubuntu WSL (as described [here](https://github.com/BetterThanTomorrow/calva/wiki/Using-with-WSL) and [here](https://github.com/BetterThanTomorrow/calva/wiki/VS-Code-Remote-Development#steps-i-took-to-try-calva-with-remote-development-in-wsl).

I hope you can help me, as I'm very eager to use you tools :)

PS: currently, I'm able to run the command `Evaluate current form in REPL terminal` with the legacy version (the description above targets the new version).

**Comment by PEZ** [comment]
Hello, and welcome to Calva! I am sorry about the troubles it gives you. It sounds like you might be using Powershell (even if that does not explain why it also fails with the Ubuntu and WSL attempts...).

There is a **Troubleshooting** section in the [Jack-in guide](https://github.com/BetterThanTomorrow/calva/wiki/About-Calva-Jack-in), which contains a suggestion for Powershell users. Can you see if that works for you?

**Comment by baflo** [comment]
Darn, I'm sure I had that set up earlier :/
Opening a REPL with Leiningen works, now... but the problem persists that I cannot connect to a ClosureScript REPL. 

If I call `Open a ClojureScript REPL` I get the error `Cannot read property 'client' of null`.

**Comment by PEZ** [comment]
Progress! :smile:

In the first Jack-in prompt, do you choose `Leiningen+Figwheel`? Assuming you do this, I'd like to know if you find any error messages in the Output channels **Calva says**, and **Calva Connection Log**.

**Comment by PEZ** [comment]
Is this still an issue for you, @baflo ?

**Comment by baflo** [comment]
Hey @PEZ, sorry it took some days... Immediately after you asked, I looked into the **Calva Connection Log**  and found that there was still a Figwheel server running on port 3449. That's why it didn't start up. When I terminated the service from task manager, the Calva REPL worked and I could evaluate from the editor. I didn't response adhoc as I wanted to check some other things. However, I didn't have time yet. 

So I just restarted the project and wanted to jack-in, but the **Calva says** log just prints `Jacking in...` and the **Calva Connection Log** keeps empty while there was a VSCode error `cannot read property 'endsWith' of undefined`.

**Comment by PEZ** [comment]
No problem. So you get this with jack-in and latest Calva? Just checking, because we've made some changes targeting windows and leiningen.

**Comment by PEZ** [comment]
@baflo , I think I found the problem causing `cannot read property 'endsWith' of undefined` with lein jack-in on windows. I just published version `2.0.14` that I hope fixes it. But I can't reproduce the bug myself, so don't really know if I succeeded. Can you give it a spin for me and tell me how it checks out?

**Comment by baflo** [comment]
Hi, I'm back from a vacation and tried the new versions of Calva. It works now! Thank you very much :)
